File: Phase_1/Leetcode/plus_one.py
"""
Input [1, 2, 3]
Output [1, 2, 4]
"""

def plus_one(digit):
    for i in range(len(digit)-1, -1, -1):
        if digit[i] < 9:
            digit[i] += 1
            return digit
        else:
            digit[i] = 0
    return [1] + digit
print(plus_one([1, 2, 3]))
print(plus_one([1, 2, 9]))

File: DSA/array-problems/plus_one.cpp
#include <iostream>
#include <vector>

using namespace std;


/*
 Problem:
 Given an array representing a number (digits), add one to the number.
 Example:
 Input: digits = [1,2,3]
 Output: [1,2,4]

 Another example (carry case):
 Input: digits = [9,9,9]
 Output: [1,0,0,0]
 */
vector<int> plus_one(vector<int>& digits){
    for (int i = digits.size() - 1; i >= 0; i--){
        if (digits[i] < 9){
            digits[i] += 1;
            return digits;
        }
        else {
            digits[i] = 0;
        }
        }

    digits.insert(digits.begin(), 1);
    return digits;
}






int main(){
    vector<int> digits = {2, 9, 9};
    plus_one(digits);
    cout << "Plus one: ";
    for (int digit: digits){
        cout << digit << " ";
    }
    cout << endl;


    return 0;
}
